### PR TITLE
Add support for setting cpu&mem limit strategy

### PR DESCRIPTION
**Pull request by mszostok**
**Description**

- Add support for setting cpu&mem limit strategy

## Demo

https://github.com/user-attachments/assets/bc74b973-c5d9-4e12-b357-a157703346a7

### CODE REVIEW

File: castai/resource_workload_scaling_policy.go [hunk 1]
```diff
@@ -203,6 +211,37 @@ func workloadScalingPolicyResourceSchema(function string, overhead, minRecommend
...
+}
+
+func workloadScalingPolicyResourceLimitSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			FieldLimitStrategyType: {
+				Type:     schema.TypeString,
+				Required: true,
+				Description: `Defines limit strategy type.
+	- NONE - removes the resource limit even if it was specified in the workload spec.
```
Comment (aivarasatkocaitiscastai) [inline castai/resource_workload_scaling_policy.go:232]: Didn't we replace `none` with `no limit`? 🤔

File: castai/resource_workload_scaling_policy.go [hunk 2]
```diff
@@ -449,10 +513,28 @@ func toWorkloadScalingPolicies(obj map[string]interface{}) sdk.Workloadoptimizat
 	if v, ok := obj["max"].(float64); ok && v > 0 {
 		out.Max = lo.ToPtr(v)
 	}
+	if v, ok := obj["limit"].([]any); ok && len(v) > 0 {
```
Comment (aivarasatkocaitiscastai) [inline castai/resource_workload_scaling_policy.go:516]: `FieldLimitStrategy` vs `"limit"`

Comment (aivarasatkocaitiscastai) [reply castai/resource_workload_scaling_policy.go:516]: I suppose here prior map access was with a const but now we have a nicely named one

File: castai/resource_workload_scaling_policy.go [hunk 3]
```diff
@@ -203,6 +211,37 @@ func workloadScalingPolicyResourceSchema(function string, overhead, minRecommend
...
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			FieldLimitStrategyType: {
+				Type:     schema.TypeString,
+				Required: true,
+				Description: `Defines limit strategy type.
+	- NONE - removes the resource limit even if it was specified in the workload spec.
+	- MULTIPLIER - used to calculate the resource limit. The final value is determined by multiplying the resource request by the specified factor.`,
+				ValidateDiagFunc: validation.ToDiagFunc(validation.StringInSlice([]string{string(sdk.MULTIPLIER), string(sdk.NOLIMIT)}, true)), // FIXME: any reason to be strict about it?
+				DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool { // FIXME:
```
[the remark this answers is not in the document]
Comment (mszostok) [reply castai/resource_workload_scaling_policy.go:235]: actually I've added this comment to make sure that I will ask about that part of the code. 
In other components, I saw that they are ignoring casing, and I agree with this approach.

However, in the policy schema, other props, e.g. `apply_type`, are case-sensitive. Is there a reason for this? I'm torn between having consistent approach and making it less error-prone 😅

File: castai/resource_workload_scaling_policy.go (hunk 2 above)
Comment (mszostok) [reply castai/resource_workload_scaling_policy.go:516]: fixed 👍

File: castai/resource_workload_scaling_policy.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (aivarasatkocaitiscastai) [reply castai/resource_workload_scaling_policy.go]: At this point I'd go with what's similar with managing the policy to not introduce more variations. Other than that I'd follow how bigger terraform providers handle such cases

**Review by aivarasatkocaitiscastai — APPROVED** [review]

**Pull request merged**